src/Tasks/Common/ConflictResolution/PackageRank.cs

```diff
@@ -16,9 +16,9 @@ public PackageRank(string[] packageIds)
 
             for (int i = numPackages - 1; i >= 0; i--)
             {
-                var preferredPackageId = packageIds[i].Trim();
+                var preferredPackageId = packageIds?[i].Trim();
 
-                if (preferredPackageId.Length != 0)
+                if (preferredPackageId is not null && preferredPackageId.Length != 0)
```

Comment:
I think you can do this?
```suggestion
                if (preferredPackageId?.Length != 0)
```

Reply:
If you change it to that, that doesn't guarantee that `preferredPackageId` is non-null. If it is null, it won't equal 0, and therefore, you couldn't use it in the if-statement, since it is used as an index. You need to check it is non-null explicitly before it is used.

Reply:
Good point. How about preferredPackageId?.Length > 0? I don't think null > 0